tofu/spectro/_rockingcurve.py

```diff
@@ -0,0 +1,345 @@
+
+
+# Built-in
+import os
+import itertools as itt
+
+# Common
+import numpy as np
+import scipy.interpolate
+import matplotlib.pyplot as plt
+import matplotlib.gridspec as gridspec
+from matplotlib.axes._axes import Axes
+from mpl_toolkits.mplot3d import Axes3D
+
+# tofu
+from tofu.version import __version__
+
+
+# ##########################################################
+# ##########################################################
+#            compute rocking curve
+# ##########################################################
+# ##########################################################
+
+
+def compute_rockingcurve(self, ih=None, ik=None, il=None, lamb=None):
+    """The code evaluates, for a given wavelength, the atomic plane distance d,
+    the Bragg angle, the complex structure factor, the integrated reflectivity
+    with the perfect and mosaic crystal models, the reflectivity curve with the
+    full dynamical model for parallel and perpendicular photon polarizations.
+
+    The alpha-Quartz, symmetry group D(4,3), is assumed left-handed.
+    It makes a difference for certain planes, for example between
+    (h,k,l)=(2,0,3) and (2,0,-3).
+    The alpha-Quartz structure is hexagonal with 3 molecules of SiO2 in the
+    unit cell.
+
+    All crystal lattice constants and wavelengths are in Angstroms (1e-10 m).
+
+    Parameters:
+    -----------
+    ih, ik, il:    int
+        Miller indices of crystal used
+    lamb:   float
+        Wavelength of interest, in Angstroms (1e-10 m)
+    """
+
+    # Check inputs
+    # ------------
+
+    lc = [ih is not None, ik is not None, il is not None, lamb is not None]
+    if any(lc) and not all(lc):
+        msg = (
+            "Args h, k, l and lamb must be provided together:\n"
+            + "\t - h: first Miller index ({})\n".format(ih)
+            + "\t - k: second Miller index ({})\n".format(ik)
+            + "\t - l: third Miller index ({})\n".format(il)
+            + "\t - lamb: wavelength of interest ({})\n".format(lamb)
+        )
+        raise Exception(msg)
```

Comment:
Good, but:

* We will probably try add more input parameters in the future
   => to be clean, move this part to a dedicated `_check_inputs()` sub-routine